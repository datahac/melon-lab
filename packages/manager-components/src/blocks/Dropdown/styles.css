.dropdown {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.dropdown__label {
  color: var(--color-grey);
  margin-right: var(--space-s);
  white-space: nowrap;
}

.dropdown__wrapper {
  position: relative;
  width: 100%;
  background-color: var(--color-border);
}

.dropdown__wrapper::before {
  content: '';
  position: absolute;
  right: 12px;
  top: var(--space-m);
  width: 0;
  height: 0;
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
  border-top: 6px solid var(--color-primary-dark);
}

.dropdown__select {
  font-family: var(--font-family-primary);
  appearance: none;
  border: none;
  width: 100%;
  background: transparent;
  font-size: var(--font-size-m);
  border-radius: 0;
  padding: var(--space-xs) var(--space-xl) var(--space-xs) var(--space-s);
  line-height: 1.6;
}

.dropdown__select::-ms-expand {
  display: none;
}

.dropdown__select:focus::-ms-value {
  background-color: transparent;
}
