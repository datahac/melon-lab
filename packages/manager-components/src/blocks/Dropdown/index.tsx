import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface DropdownProps {
  options: Array<{
    value: string;
    name: string;
  }>;
  label: string;
  value?: string;
  name: string;
  disabled: boolean;
  onChange: (value, event) => void;
}

const renderOption = (option, index) => {
  return (
    <option key={index} value={option.value}>
      {option.name}
    </option>
  );
};

const Dropdown: StatelessComponent<DropdownProps> = ({
  options,
  onChange,
  label,
  value,
  name,
  disabled,
}) => {
  const ops =
    options && options.map((item, index) => renderOption(item, index));

  const handleChange = (e: any): void => {
    const targetValue: any = e.target.value;
    onChange({ value: targetValue }, e);
  };

  return (
    <div className="dropdown">
      <style jsx>{styles}</style>
      {label && <span className="dropdown__label">{label}</span>}
      <div className="dropdown__wrapper">
        <select
          name={name}
          className="dropdown__select"
          onChange={handleChange}
          value={value}
          disabled={disabled}
        >
          {ops}
        </select>
      </div>
    </div>
  );
};

export default Dropdown;
