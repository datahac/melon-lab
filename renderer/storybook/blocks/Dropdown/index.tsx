import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface DropdownProps {
  disabled?: boolean;
  label?: string;
  name: string;
  onChange: (e) => void;
  options: Array<{
    value: string;
    name: string;
  }>;
  value?: string;
}

const Dropdown: StatelessComponent<DropdownProps> = ({
  disabled,
  label,
  name,
  onChange,
  options,
  value,
}) => {
  return (
    <div className="dropdown">
      <style jsx>{styles}</style>
      {label && <span className="dropdown__label">{label}</span>}
      <div className="dropdown__wrapper">
        <select
          name={name}
          className="dropdown__select"
          onChange={onChange}
          value={value}
          disabled={disabled}
        >
          {options &&
            options.map((item, index) => (
              <option key={index} value={item.value}>
                {item.name}
              </option>
            ))}
        </select>
      </div>
    </div>
  );
};

export default Dropdown;
