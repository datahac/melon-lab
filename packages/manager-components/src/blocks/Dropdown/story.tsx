import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Dropdown from './index';

const data = {
  name: 'name';
  options: [{ value: 'one', name: 'One' }, { value: 'two', name: 'Two' }],
  onChange: action('clicked'),
};

storiesOf('Blocks|Dropdown', module)
  .add('Default', () => {
    return <Dropdown {...data} />;
  })
  .add('With label', () => {
    return <Dropdown {...data} label="Exchange" />;
  });
