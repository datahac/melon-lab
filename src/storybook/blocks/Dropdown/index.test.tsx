import React from 'react';
import Dropdown from './index';

const data = {
  name: 'name',
  label: 'label',
  options: [{ value: 'one', name: 'One' }, { value: 'two', name: 'Two' }],
  onChange: () => null,
};

describe('Dropdown', () => {
  const defaultElement = <Dropdown {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass a selected value to the onChange handler', () => {
    const onChange = jest.fn();
    wrapper.setProps({ onChange });

    expect(wrapper).toMatchSnapshot();
    wrapper.find('select').simulate('change', { target: { value: 'value' } });
    expect(onChange).toBeCalledWith({ target: { value: 'value' } });
  });
});
