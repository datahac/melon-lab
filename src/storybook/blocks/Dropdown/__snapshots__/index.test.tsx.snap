// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should pass a selected value to the onChange handler 1`] = `
<div
  className="dropdown"
>
  <style
    jsx={true}
  >
    <type
      0="p"
      1="r"
      2="o"
      3="p"
      4="s"
      key="key"
    >
      rendered
    </type>
  </style>
  <span
    className="dropdown__label"
  >
    label
  </span>
  <div
    className="dropdown__wrapper"
  >
    <select
      className="dropdown__select"
      name="name"
      onChange={[MockFunction]}
    >
      <option
        key="0"
        value="one"
      >
        One
      </option>
      <option
        key="1"
        value="two"
      >
        Two
      </option>
    </select>
  </div>
</div>
`;

exports[`Dropdown should render correctly 1`] = `
<div
  className="dropdown"
>
  <style
    jsx={true}
  >
    <type
      0="p"
      1="r"
      2="o"
      3="p"
      4="s"
      key="key"
    >
      rendered
    </type>
  </style>
  <span
    className="dropdown__label"
  >
    label
  </span>
  <div
    className="dropdown__wrapper"
  >
    <select
      className="dropdown__select"
      name="name"
      onChange={[Function]}
    >
      <option
        key="0"
        value="one"
      >
        One
      </option>
      <option
        key="1"
        value="two"
      >
        Two
      </option>
    </select>
  </div>
</div>
`;
